moby/moby
# awslogs: Update aws-sdk-go to support IMDSv2

**- What I did**
AWS recently launched a new version of the EC2 Instance Metadata Service, which is used to provide credentials to the awslogs driver when running on Amazon EC2.  This new version of the IMDS adds defense-in-depth mechanisms against open firewalls, reverse proxies, and SSRF vulnerabilities and is generally an improvement over the previous version.  An updated version of the AWS SDK is able to handle the both the previous version and the new version of the IMDS and functions when either is enabled.

More information about IMDSv2 is available at the following links:

* https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/
* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

Closes https://github.com/moby/moby/issues/40422

**- How I did it**
* Updated the dependency on github.com/aws/aws-sdk/go
* Updated the transitive dependency on github.com/jmespath/go-jmespath
* Removed unused dependency github.com/go-ini/ini

**- How to verify it**
1. Run an EC2 instance with IMDSv1 disabled.  This can be set at launch time via a parameter to the RunInstances API, or can be set after an instance has already launched via the ModifyInstanceMetadataOptions API.
1. Ensure that the instance has an IAM profile allowing it to write to CloudWatch Logs.
1. Install this updated version of Docker
1. Run a container with the `awslogs` driver and ensure that it writes to CloudWatch Logs successfully

**- Description for the changelog**

awslogs: Update aws-sdk-go to support IMDSv2
